adapter.go

```diff
@@ -826,11 +926,11 @@ func (a *Adapter) UpdateFilteredPolicies(sec string, ptype string, newPolicies [
 			tx.Rollback()
 			return nil, err
 		}
-		if err := tx.Where(str, args...).Delete([]CasbinRule{}).Error; err != nil {
+		if err := a.dbDelete(tx.Where(str, args...)); err != nil {
 			tx.Rollback()
 			return nil, err
 		}
-		if err := tx.Create(&newP[i]).Error; err != nil {
+		if err := a.dbCreate(tx, &newP[i]); err != nil {
```

Comment:
Please check if these two methods should be placed outside the ``for`` loop and executed only once rather than repeating.